From cc7a0fc21a40bf614443ecc64626f67dfcb98098 Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Tue, 16 Oct 2018 09:59:35 +0300
Subject: [PATCH 2/2] arm64: dts: am654-base-board: add wl18xx support

the am654 base board doesn't have a com8 connector for conencting
a standard wl18xx com8 module so a wl118xx module can be connected to the
evm only using a com2sdmmc adapter board plugged into the SD Card slot.
As a result boot-images/root-filesystem have to move to emmc.

Additional IOs are required for wl18xx control pins (wlan_enable,
wlan_irq and bluetooth_enable). the SPI1 header (j20) of the base board
is used for connecting these IOs to the wl18xx adapter board.

Lastly, the mcu uart1 header (J30) is used for uart connection to the
adapter board (bluetooth subsystem of wl18xx).

Signed-off-by: Eyal Reizer <eyalr@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am654-base-board.dts | 69 +++++++++++++++++++++++++-
 1 file changed, 68 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am654-base-board.dts b/arch/arm64/boot/dts/ti/k3-am654-base-board.dts
index c1d4d39..c94aa68 100644
--- a/arch/arm64/boot/dts/ti/k3-am654-base-board.dts
+++ b/arch/arm64/boot/dts/ti/k3-am654-base-board.dts
@@ -126,6 +126,20 @@
 			gpios = <&wkup_gpio0 27 GPIO_ACTIVE_LOW>;
 		};
 	};
+
+	wlan_en_reg: fixedregulator {
+		compatible = "regulator-fixed";
+		regulator-name = "wlan-en-regulator";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		/* WLAN_EN GPIO for this board - Bank1, pin23 */
+		gpio = <&main_gpio1 23 0>;
+
+		/* WLAN card specific delay */
+		startup-delay-us = <70000>;
+		enable-active-high;
+	};
 };
 
 &mcu_r5f0 {
@@ -230,6 +244,19 @@
 			AM65X_IOPAD(0x0088, PIN_INPUT | MUX_MODE2) /* (AG17) PRG2_PRU0_GPO4.PRG2_RGMII1_RX_CTL */
 		>;
 	};
+
+	main_wlan_pins_default: main_wlan_pins_default {
+		pinctrl-single,pins = <
+			AM65X_IOPAD(0x01dc, PIN_OUTPUT_PULLDOWN | MUX_MODE7) /* (AE12) GPIO1_23 */
+			AM65X_IOPAD(0x01e0, PIN_INPUT | MUX_MODE7) /* (AF12) GPIO1_24 */
+		>;
+	};
+
+	main_bt_pins_default: main_bt_pins_default {
+		pinctrl-single,pins = <
+			AM65X_IOPAD(0x01d8, PIN_OUTPUT | MUX_MODE7) /* (AH12) GPIO1_22 */
+		>;
+	};
 };
 
 &main_pmx1 {
@@ -314,6 +341,15 @@
 			AM65X_WKUP_IOPAD(0x0088, PIN_INPUT | MUX_MODE0) /* (L4) MCU_MDIO0_MDIO */
 		>;
 	};
+
+	mcu_uart0_pins_default: mcu_uart0_pins_default {
+		pinctrl-single,pins = <
+			AM65X_WKUP_IOPAD(0x0044, PIN_INPUT | MUX_MODE4)  /* (P4) MCU_OSPI1_D1.MCU_UART0_RXD */
+			AM65X_WKUP_IOPAD(0x0048, PIN_OUTPUT | MUX_MODE4) /* (P5) MCU_OSPI1_D2.MCU_UART0_TXD */
+			AM65X_WKUP_IOPAD(0x004C, PIN_INPUT | MUX_MODE4)  /* (P1) MCU_OSPI1_D3.MCU_UART0_CTSn */
+			AM65X_WKUP_IOPAD(0x0054, PIN_OUTPUT | MUX_MODE4) /* (N3) MCU_OSPI1_CSn1.MCU_UART0_RTSn */
+		>;
+	};
 };
 
 &wkup_i2c0 {
@@ -352,6 +388,12 @@
 		reg = <0x21>;
 		gpio-controller;
 		#gpio-cells = <2>;
+		p1 {
+			gpio-hog;
+			gpios = <14 GPIO_ACTIVE_HIGH>;
+			output-low;
+			line-name = "uart_sel";
+		};
 	};
 };
 
@@ -443,7 +485,21 @@
 &sdhci1 {
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&main_mmc1_pins_default>;
+	vmmc-supply = <&wlan_en_reg>;
+	pinctrl-0 = <&main_mmc1_pins_default &main_wlan_pins_default>;
+	non-removable;
+	cap-power-off-card;
+	keep-power-in-suspend;
+	max-frequency = <24000000>;
+
+	#address-cells = <1>;
+	#size-cells = <0>;
+	wlcore: wlcore@0 {
+		compatible = "ti,wl1835";
+		reg = <2>;
+		interrupt-parent = <&main_gpio1>;
+		interrupts = <24 IRQ_TYPE_EDGE_RISING>;
+	};
 };
 
 &mmc_phy1 {
@@ -490,3 +546,14 @@
 		ti,adc-channels = <0 1 2 3 4 5 6 7>;
 	};
 };
+
+&mcu_uart0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&mcu_uart0_pins_default &main_bt_pins_default>;
+	bluetooth {
+		compatible = "ti,wl1837-st";
+		enable-gpios = <&main_gpio1 22 GPIO_ACTIVE_HIGH>;
+		max-speed = <3000000>;
+	};
+};
-- 
2.7.4

